### PR TITLE
chore(dev): Detail the format of DEPRECATIONS.md file

### PR DESCRIPTION
**Pull request by jszwedko**
The format of this file was originally described in https://github.com/vectordotdev/vector/issues/14983

Signed-off-by: Jesse Szwedko <jesse.szwedko@datadoghq.com>

### CODE REVIEW

File: docs/DEPRECATIONS.md [hunk 1]
```diff
@@ -1,5 +1,13 @@
 See [DEPRECATION.md](docs/DEPRECATION.md#process) for the process for updating this file.
 
+The format for each entry should be: `<version> <identifier> <description>`.
```
Comment (neuronull) [inline docs/DEPRECATIONS.md:3]: It might be nice to add an example

**Review by neuronull — APPROVED** [review]

File: docs/DEPRECATIONS.md (hunk 1 above)
Comment (jszwedko) [reply docs/DEPRECATIONS.md:3]: Good call, added.

**Review by bruceg — APPROVED** [review]

File: docs/DEPRECATIONS.md [hunk 2]
```diff
@@ -1,11 +1,23 @@
...
+
+- `<version>` should be the version of Vector in which to take the action (deprecate, migrate, or
+  remove)
+- `<identifier>` should be a unique identifier that can also be used in the code to easily find the
+  places to modify
+- `<description>` should be a longer form description of the change to be made
+
+For example:
+
+- legacy_openssl_provider v0.34.0 OpenSSL legacy provider flag should be removed
```
Comment (dsmith3197) [inline docs/DEPRECATIONS.md:13]: The version and identifier should be swapped right?
```suggestion
- v0.34.0 legacy_openssl_provider OpenSSL legacy provider flag should be removed
```

**Review by dsmith3197 — APPROVED** [review]
Looks good, I'll update https://github.com/vectordotdev/vector/pull/18972 to adhere to the format specified 🙂

**Pull request merged**